tests/admin/valuesets/valueSetCreation.spec.ts

```diff
@@ -0,0 +1,347 @@
+    await page.getByRole("option", { name: "LOINC" }).click();
+
+    // Add multiple concepts
+    await page.getByRole("button", { name: "+ Add Concepts" }).first().click();
+    await page.getByRole("textbox", { name: "Code *" }).first().fill("1558-6");
+
+    await page.getByRole("button", { name: "+ Add Concepts" }).first().click();
+    const secondConcept = page.getByRole("textbox", { name: "Code *" }).nth(1);
+    await secondConcept.fill("2339-0");
+
+    // Save and verify
+    await page.getByRole("button", { name: "Save" }).click();
+
+    await expect(page.getByText("ValueSet created successfully")).toBeVisible({
+      timeout: 10000,
+    });
+    await expect(page.getByRole("cell", { name: uniqueName })).toBeVisible();
+  });
+
+  test("should create valueset with include rules - any system with concept only", async ({
+    page,
+  }) => {
+    const uniqueSuffix = faker.string.alphanumeric(8).toLowerCase();
+    const uniqueName = `Concept Only ValueSet ${uniqueSuffix}`;
+    const uniqueSlug = `concept-only-${uniqueSuffix}`;
+
+    // Fill basic info
+    await page.getByRole("textbox", { name: "Name *" }).fill(uniqueName);
+    await page.getByRole("textbox", { name: "Slug *" }).fill(uniqueSlug);
+
+    // Add include rule with SNOMED concept
+    await page
+      .getByRole("button", { name: "+ Add Rule", exact: true })
+      .first()
+      .click();
+    await page.getByRole("combobox", { name: "System" }).first().click();
+    await page.getByRole("option", { name: "SNOMED CT" }).click();
+
+    // Add concept and verify it's properly added
+    await page.getByRole("button", { name: "+ Add Concepts" }).first().click();
+    const conceptCode = "73211009"; // Diabetes mellitus SNOMED code
+    await page
+      .getByRole("textbox", { name: "Code *" })
+      .first()
+      .fill(conceptCode);
+
+    // Set status
+    await page.getByRole("combobox", { name: "Status" }).click();
+    await page.getByRole("option", { name: "Active" }).click();
+
+    // Verify concept is properly set before saving
+    await expect(
+      page.getByRole("textbox", { name: "Code *" }).first(),
+    ).toHaveValue(conceptCode);
+
+    // Test ValueSet Preview
+    await page.getByRole("button", { name: "Value Preview" }).click();
+
+    // Verify preview dialog opened
+    const previewDialog = page
+      .locator('[data-testid="preview-dialog"]')
+      .first();
+    await expect(previewDialog).toBeVisible();
+
+    // Verify specific codes appear in preview
+    await expect(previewDialog.getByText(conceptCode)).toBeVisible();
+
+    // Close preview and save
+    await previewDialog.getByRole("button", { name: /close|back/i }).click();
+    await page.getByRole("button", { name: "Save" }).click();
+
+    // Verify success
+    await expect(page.getByText("ValueSet created successfully")).toBeVisible({
+      timeout: 10000,
+    });
+  });
+
+  test("should create valueset with include rules - any system with filters", async ({
```

Comment:
- for each system you need to add the separate test, and make all the necessary changes here as well along along with its visibility in valueset preview